classes/issue/DAO.inc.php

```diff
@@ -0,0 +1,484 @@
+        'dateNotified' => 'date_notified',
+        'lastModified' => 'last_modified',
+        'accessStatus' => 'access_status',
+        'openAccessDate' => 'open_access_date',
+        'showVolume' => 'show_volume',
+        'showNumber' => 'show_number',
+        'showYear' => 'show_year',
+        'showTitle' => 'show_title',
+        'styleFileName' => 'style_file_name',
+        'originalStyleFileName' => 'original_style_file_name',
+        'urlPath' => 'url_path'
+    ];
+
+    /**
+     * Handle a cache miss.
+     *
+     * TODO: Caching not currently working as expected
+     *
+     */
+    public function _cacheMiss(GenericCache $cache, int $id): ?Issue
+    {
+        if ($cache->getCacheId() === 'current') {
+            $issue = Repo::issue()->getCurrent($id);
+        } else {
+            $issue = Repo::issue()->getByBestId($id, null, false);
+        }
+        $cache->setCache($id, $issue);
+        return $issue;
+    }
+
+    /**
+     * Get an issue cache by cache ID
+     *
+     * TODO: Not currently working as expected. Not used throughout current class
+     *
+     * @return mixed|object|\PKP\cache\APCCache|\PKP\cache\FileCache|GenericCache|\PKP\cache\MemcacheCache|\PKP\cache\XCacheCache
+     */
+    public function _getCache(string $cacheId)
+    {
+        if (!isset($this->caches)) {
+            $this->caches = [];
+        }
+        if (!isset($this->caches[$cacheId])) {
+            $cacheManager = CacheManager::getManager();
+            $this->caches[$cacheId] = $cacheManager->getObjectCache('issues', $cacheId, [$this, '_cacheMiss']);
+        }
+        return $this->caches[$cacheId];
+    }
+
+    /**
+     * Instantiate a new DataObject
+     */
+    public function newDataObject(): Issue
+    {
+        return App::make(Issue::class);
+    }
+
+    /**
+     * Check if a submission exists
+     */
+    public function exists(int $id): bool
+    {
+        return DB::table($this->table)
+            ->where($this->primaryKeyColumn, '=', $id)
+            ->exists();
+    }
+
+    /**
+     * @copydoc EntityDAO::get()
+     */
+    public function get(int $id): ?Issue
+    {
+        return parent::get($id);
+    }
+
+    /**
+     * Get the number of announcements matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('i.' . $this->primaryKeyColumn)
+            ->pluck('i.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of issues matching the configured query
+     *
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->get();
```

Comment:
:+1: That's great! I'm hoping that the new structure provides room for these kinds of performance improvements.

The `LazyCollection` here is still useful because this `get()` only returns the row data. The `fromRow` method, which instantiates the `Issue` objects and often runs additional queries), won't be called until the collection is looped over.